src/main.rs

```diff
@@ -32,17 +40,33 @@ struct ComponentCode {
     code: String,
 }
 
-fn handle_request(account_id: &str, path: PathBuf) -> FileList {
+fn handle_request(account_id: &str, path: PathBuf, network: Network) -> FileList {
     let mut components = HashMap::new();
-    get_file_list(&path, account_id, &mut components, String::from(""));
+    get_file_list(&path, account_id, &mut components, String::from(""), network);
     FileList { components }
 }
 
+fn replace_placeholders(code: &str, account_id: &str, network: Network) -> String {
+    let mut replacements = HashMap::new();
+    replacements.insert("${REPL_ACCOUNT}", account_id);
+    replacements.insert("${REPL_NEAR_URL}", if network == Network::Testnet {"test.near.org"} else {"near.org"});
```

Comment:
this should be more generic because `bos-loader` is used by developers across gateways. We probably either want an additional flag to specify gateway URL directly or want to define a full map of replacements. I have another WIP branch which adds a configuration file for more complex setups so let's pause until I tie that up because they will likely play nice with each other

Reply:
What do you think of adding an optional arg that would specify a path to a replacements file? If no path is specified, the above map is used. If the user passes a file path, its content should be merged with the map above. The file could have the same format as the one used [in here](https://github.com/near/near-discovery-components/pull/107/files#diff-fefbbfc7ff66fae66581bec368540ca31b7f5dfca81e2ae4c68248fb6ab3afa1) which gives the user a possibility to just use the same file in both places. The downside of this approach is that the network parameter needs to align with the network names found in replacements.json.
The map of replacements could be also specified in the config file as you described above. Then all the config-related info is found in the same spot which is also a big advantage.